src/components/Notification/notifyTable.tsx

```diff
@@ -51,25 +50,26 @@ const truncateText = (text: string, maxLength: number) => {
 };
 
 const NotifyTable: React.FC = () => {
-  const [userData, setUserData] = useState<UserData[]>([]);
+  const [userData, setUserData] = useState<UserData[]>([]); // Initialize as an empty array
   const [loading, setLoading] = useState(true);
   const { toast } = useToast();
-  
-    const fetchUserData = async () => {
-      try {
-        const response = await apiHelperService.getAllNotification();
-        setUserData(response.data.data);
-      } catch (error) {
-        toast({
-          title: "Error",
-          description: Messages.FETCH_ERROR("notification"),
-          variant: "destructive", 
-        });
-      } finally {
-        setLoading(false);
-      }
-    };
-    useEffect(() => {
+
+  const fetchUserData = async () => {
+    try {
+      const response = await apiHelperService.getAllNotification();
+      setUserData(response.data.data || []); // Default to an empty array if data is undefined
```

Comment:
your current code assumes the presence of response.data.data, which might lead to a runtime error if the structure of response or its nested properties (data or data.data) doesn't exist as expected

add null check in response to prevent this